packages/selected-network-controller/src/SelectedNetworkController.ts

```diff
@@ -311,6 +307,10 @@ export class SelectedNetworkController extends BaseController<
     domain: Domain,
     networkClientId: NetworkClientId,
   ) {
+    if (!this.#useRequestQueuePreference) {
+      return;
+    }
```

Comment:
nit this could be bundled with the snaps check below which returns 
```
   if (snapsPrefixes.some((prefix) => domain.startsWith(prefix)) || !this.#useRequestQueuePreference) {
      return;
    }
```